raiden/raiden_service.py

```diff
@@ -866,41 +852,37 @@ def transfer_async(self, token_address, amount, target, identifier=None):
 
         return async_result
 
-    def _direct_or_mediated_transfer(self, token_address, amount, identifier, direct_channel):
-        """ Check the direct channel and if possible use it, otherwise start a
-        mediated transfer.
-        """
+    def direct_transfer_async(self, token_address, amount, target, identifier):
+        """ Do a direct tranfer with target.
 
-        if not direct_channel.can_transfer:
-            log.info(
-                'DIRECT CHANNEL is closed or has no funding',
-                from_=pex(direct_channel.our_state.address),
-                to=pex(direct_channel.partner_state.address),
-            )
+        Direct transfers are non cancellable and non expirable, since these
+        transfers are a signed balance proof with the transferred amount
+        incremented.
 
-            async_result = self._mediated_transfer(
-                token_address,
-                amount,
-                identifier,
-                direct_channel.partner_state.address,
-            )
+        Because the transfer is non cancellable, there is a level of trust with
+        the target, after the message is sent the target is effectively paid
+        and there it is not possible to revert.
 
-        elif amount > direct_channel.distributable:
-            log.info(
-                'DIRECT CHANNEL does not have enough funds',
-                from_=pex(direct_channel.our_state.address),
-                to=pex(direct_channel.partner_state.address),
-                amount=amount,
-            )
+        The async result will be set to False iif there is no direct channel
```

Comment:
Did you want to say `iif` -> `iff` ? Meaning "If and only if"?

Reply:
I meant "if and only if"

Reply:
yeah figured as much :)